app/controllers/samples_controller.rb

```diff
@@ -800,19 +800,39 @@ def samples_going_public
 
   def report_v2
     pipeline_run = select_pipeline_run(@sample, params[:pipeline_version])
-    if pipeline_run
-      background_id = get_background_id(@sample, params[:background])
-      render json: PipelineReportService.call(pipeline_run.id, background_id)
-    else
-      render json: {
-        metadata: {
-          pipelineRunStatus: "WAITING",
-          jobStatus: "Waiting to Start or Receive Files",
-          errorMessage: nil,
-          knownUserError: nil,
-        },
-      }
-    end
+    background_id = get_background_id(@sample, params[:background])
+
+    skip_cache = params[:skip_cache] || false
+    MetricUtil.log_analytics_event("samples.cache.requested", current_user) unless skip_cache
+    report_info_params = pipeline_run.report_info_params
+
+    cache_key = PipelineReportService.report_info_cache_key(
+      request.path,
+      params
+        .permit(report_info_params.keys)
+        .merge(pipeline_run_id: pipeline_run.id)
+    )
+
+    json =
```

Comment:
It feels hard to read this conditional assignment with so much in the `else` block. did Rubocop force this? Maybe have it as something like `unless skip_cache, compute all the headers`?

Reply:
At @tfrcarvalho 's request, I pulled the cache logic out into its own function so the block should be cleaner now!